api/api/prediction_job_api.go

```diff
@@ -39,10 +38,10 @@ func (c *PredictionJobController) Create(r *http.Request, vars map[string]string
 
 	model, version, err := c.getModelAndVersion(ctx, modelID, versionID)
 	if err != nil {
-		if !gorm.IsRecordNotFoundError(err) {
-			return InternalServerError(err.Error())
+		if gorm.IsRecordNotFoundError(err) {
+			return NotFound(fmt.Sprintf("Model / version not found: %v", err))
```

Comment:
Whoa thanks for catching the misplaced `!`! Ditto for the change below (in the same file).